pkg/config/api/config.go

```diff
@@ -0,0 +1,46 @@
+package api
+
+type Config struct {
+	Components Components `json:components`
+}
+
+type Components struct {
+	Network Network `json:network`
+	DNS     DNS     `json:dns`
+}
+
+type Network struct {
+	// IP address pool to use for pod IPs.
+	// This field is immutable after installation.
+	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork"`
+
+	// IP address pool for services.
+	// Currently, we only support a single entry here.
+	// This field is immutable after installation.
+	ServiceNetwork []string `json:"serviceNetwork"`
+
+	// The port range allowed for Services of type NodePort.
+	// If not specified, the default of 30000-32767 will be used.
+	// Such Services without a NodePort specified will have one
+	// automatically allocated from this range.
+	// This parameter can be updated after the cluster is
+	// installed.
+	// +kubebuilder:validation:Pattern=`^([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])-([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$`
+	ServiceNodePortRange string `json:"serviceNodePortRange,omitempty"`
```

Comment:
are we planning to add/support `ExternalIPNetworkCIDRs` for LoadBalancer Services? 

https://access.redhat.com/documentation/en-us/openshift_container_platform/3.3/html/cluster_administration/admin-guide-unique-external-ips-ingress-traffic

Reply:
in OpenShift 4 configured via `networks.config/cluster` and `.spec.externalIP.AutoAssignCIDRs` field

https://access.redhat.com/documentation/en-us/openshift_container_platform/4.11/html-single/networking/index#nw-service-externalip-create_configuring-ingress-cluster-traffic-service-external-ip

Reply:
We model the Network part of the config using [Network](https://pkg.go.dev/github.com/openshift/api/config/v1#NetworkSpec) type in OS config v1. That type spec does have `ExternalIP` config which we can use in the future if there is a need to support this feature in the future.
For now, we just rework the existing fields in the config into a new config that is aligned with OS config v1 API.

Reply:
@zshi-redhat or @mangelajo do either of you have an opinion about @atiratree's question about ExternalIPNetworkCIDR?

Reply:
@dhellmann sorry I missed this question.

Yes, ovn-kubernetes supports k8s service of ExternalIP type and it can be used today with prerequisites that the network infrastructure (outside MicroShift cluster) needs to route traffic for the external IP address to MicroShift cluster. However, we probably don't need `ExternalIPNetworkCIDRs` or `AutoAssignCIDRs` since these two API fields are exposed by OpenShift operators which are not supposed to run in regular MicroShift deployment.
So we don't have external IP pool management in MicroShift due to lack of operators support, but user can attach the selected external IP manually when creating/patching the k8s service.

Reply:
in that case I have created a PR that disables the controller so it doesn't have to run https://github.com/openshift/microshift/pull/1058